easybuild/easyblocks/c/comsol.py

```diff
@@ -0,0 +1,136 @@
+            lic_specs=[self.cfg['license_file']])
+
+        if lic_specs:
+            if self.license_env_var is None:
+                self.log.info("Using Comsol license specifications from 'license_file': %s", lic_specs)
+                self.license_env_var = default_lic_env_var
+            else:
+                self.log.info("Using Comsol license specifications from $%s: %s", self.license_env_var, lic_specs)
+
+            self.license_file = os.pathsep.join(lic_specs)
+            env.setvar(self.license_env_var, self.license_file)
+        else:
+            msg = "No viable license specifications found; "
+            msg += "specify 'license_file', or define $LM_LICENSE_FILE"
+            raise EasyBuildError(msg)
+
+        try:
+            shutil.copyfile(os.path.join(self.cfg['start_dir'], 'setupconfig.ini'), self.configfile)
+            config = read_file(self.configfile)
+
+            regdest = re.compile(r"^installdir\s*=.*", re.M)
+            reggui = re.compile(r"^showgui\s*=.*", re.M)
+            regagree = re.compile(r"^agree\s*=.*", re.M)
+            reglicpath = re.compile(r"^license\s*=.*", re.M)
+            reglinuxlauncher = re.compile(r"^linuxlauncher\s*=.*", re.M)
+            regsymlinks = re.compile(r"^symlinks\s*=.*", re.M)
+            regfileassoc = re.compile(r"^fileassoc\s*=.*", re.M)
+
+            config = regdest.sub("installdir=%s" % self.installdir, config)
+            config = reggui.sub("showgui=0", config)
+            config = regagree.sub("agree=1", config)
+            config = reglicpath.sub("license=%s" % self.license_file, config)
+            config = reglinuxlauncher.sub("linuxlauncher=0", config)
+            config = regsymlinks.sub("symlinks=0", config)
+            config = regfileassoc.sub("fileassoc=0", config)
+
+            matlab_root = get_software_root("MATLAB")
+            if matlab_root:
+                regmatlab = re.compile(r"^matlabdir\s*=.*", re.M)
+                config = regmatlab.sub("matlabdir=%s" % matlabroot, config)
+
+            write_file(self.configfile, config)
+
+        except IOError, err:
+            raise EasyBuildError("Failed to create installation config file %s: %s", self.configfile, err)
+
+        self.log.debug('configuration file written to %s:\n %s', self.configfile, config)
+
+    def install_step(self):
+        """COMSOL install procedure using 'install' command."""
+
+        src = os.path.join(self.cfg['start_dir'], 'setup')
+
+        # make sure setup script is executable
+        adjust_permissions(src, stat.S_IXUSR)
+
+        # make sure $DISPLAY is not defined, which may lead to (hard to trace) problems
+        # this is a workaround for not being able to specify --nodisplay to the install scripts
+        if 'DISPLAY' in os.environ:
+            os.environ.pop('DISPLAY')
+
+        cmd = "%s ./setup -s %s %s" % (self.cfg['preinstallopts'], self.configfile, self.cfg['installopts'])
```

Comment:
Why not use `src` here rather than hardcoding `setup` again?

```python
cmd = ' '.join([self.cfg['preinstallopts'], src, '-s', self.configfile, self.cfg['installopts']])